source/types/acorn-jsx.d.ts

```diff
@@ -17,6 +17,22 @@ declare module 'acorn-jsx' {
 		argument?: Expression;
 	}
 
+	interface BaseFunctionExpression extends BaseExpression {
+		async: Boolean
+		generator: Boolean
+		expression: true;
+		argument?: Expression;
+		body: Expression
+		params: Identifier[]
+	}
+
+  export interface ArrowFunctionExpression extends BaseFunctionExpression {
+		type: 'ArrowFunctionExpression';
+	}
+	export interface FunctionExpression extends BaseFunctionExpression {
```

Comment:
I left this in pending discussion around support for "full" functions (e.g. arrow with braces/return and "normal" function expression). 

I can remove this if we decide simple arrows are the way to go.

Reply:
My vote is that simple arrows are the way to go because it keeps things simpler.